server/src/oauth/google_login.rs

```diff
@@ -0,0 +1,76 @@
+use super::grant_authorization_code;
+use super::verify_oauth_query;
+use super::AuthorizationRequestQuery;
+use crate::State;
+use axum::extract::Extension;
+use axum::extract::Form;
+use axum::extract::Query;
+use axum::extract::TypedHeader;
+use headers::Cookie;
+use houseflow_types::errors::AuthError;
+use houseflow_types::errors::InternalError;
+use houseflow_types::errors::OAuthError;
+use houseflow_types::errors::ServerError;
+use jsonwebtoken_google::Parser;
+use serde::Deserialize;
+use serde::Serialize;
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct Request {
+    credential: String,
+    g_csrf_token: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct TokenClaims {
```

Comment:
maybe we could use existing JWT implementation in houseflow_types::token, it is compliant with the JWT standard, https://github.com/gbaranski/houseflow/blob/master/types/src/token.rs.

Reply:
Your implementation there is missing a bunch of features that are needed for this (support for asymmetric signatures like RS256, more validation options). On top of that, the `jsonwebtoken-google` handles fetching and caching the appropriate public keys from Google to do the signature verification. I'd really rather not reimplement it all from scratch, given that someone has already made a nice library to handle it all. In general, isn't it better to write less code, so you have less to maintain and hopefully fewer bugs?

Actually, my next PR was going to be replacing most of the implementation in `token.rs` with a thin wrapper around `jsonwebtoken` to simplify the code. I have that working already, I just need to test it a bit more before I send the PR.

Reply:
I wrote `token.rs` because I had some issues with `jsonwebtoken` crate. But if you think that it'll work better than my own implementation, that's fine.

Reply:
It definitely works well for this case of handling Google login tokens. I think it also works well to replace your implementation, I've sent https://github.com/gbaranski/houseflow/pull/236 to do that if you want to try it out.